libs/cart/src/cart-state.module.ts

```diff
@@ -15,7 +15,7 @@ import { DaffCartPaymentMethodsEffects } from './effects/cart-payment-methods.ef
 
 @NgModule({
   imports: [
-    StoreModule.forFeature('cart', daffCartReducers),
+    StoreModule.forFeature('cart', daffCartReducers()),
```

Comment:
You made `daffCartReducers` generic but how would a type parameter be passed into it?

Reply:
The way I understand it, it all comes together through the driver. If you don't provide a correct DaffCartDriver that extends the DaffCartServiceInterface, then you should get a build error (not sure it actually does right now though, because injection tokens can't be typed yet). Then that makes it incompatible with the effects layer, which makes it incompatible with the reducers layer. I'm not super solid on how exactly everything with generics work, but it seems like when an instance of something doesn't care about what the concrete object is, it's ok to stay generic. Here, the daffCartReducers() function doesn't require a type other than `T extends DaffCart`, so the generic version is enough.